### PR TITLE
Free flow: Add design selection screen

### PR DESCRIPTION
**Pull request by agrullon95**
#### Proposed Changes

* Changes to use `processing` step (loading screen) twice for free flow.
The first loading screen appears after the site setup screen (background process for creating the new site)
The second loading screen appears after the design selection screen (background process for activating theme on site)

You can ignore styling issues since this is for testing the functionality of the flow.

#### Testing Instructions

* Checkout this PR
* Navigate to `/setup/free/intro`
* Go through the flow, select a free theme during the design selection step and continue through the flow until you land on the launchpad with a new site.
* Go through the flow, select a paid theme during the design selection step and continue through the flow until you land on the launchpad with a new site.

image

image

image

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #70559

### CODE REVIEW

**Comment by sharpirate** [comment]
✔️ This is testing well for me.

One thing to note is when using the calypso live link and a premium theme I got an infinite loading screen at the end because the `?flags=signup/free-flow` flag is being stripped from the URL while going through the purchase screen. But it works well when testing on local calypso.

**Review by sharpirate — APPROVED** [review]

**Comment by edanzer** [comment]
This tests as expected, for both free theme and premium themes (which include a checkout process). 

A few small notes/questions:

1. We're no longer using the 'patterns' step. I think we can go ahead and remove that from the list of steps?
2. I know it's not part of this PR, but I also see the 'domains' and 'plans' in the list of steps for the flow. We probably should have removed those as part of https://github.com/Automattic/wp-calypso/pull/71256. Assuming we're going to remove the patterns step in this PR, it may be worth removing the domains and plans steps here too, as those screen are not used any more. 
3. I see my domain is being set to [username][number].wordpress.com when I arrive on the design selection screen, and later on launchpad. I believe that's correct and is now the intended behavior now following the merge of the PR above. 
4. I know you said not to worry about styling, but I figured it may be worth noting issues for now or to handle later. Without checking too carefully, I see the back button is still showing and overlaps the Free page title. Also the progress bar at the top has some space above it, and also doesn't seem to be tracking progress (it is all gray). Are we planning to address styling issues in a later PR?

**Comment by agrullon95** [comment]
@edanzer 

> 1 .We're no longer using the 'patterns' step. I think we can go ahead and remove that from the list of steps?

I removed these steps.

> 2. I know it's not part of this PR, but I also see the 'domains' and 'plans' in the list of steps for the flow. We probably should have removed those as part of https://github.com/Automattic/wp-calypso/pull/71256. Assuming we're going to remove the patterns step in this PR, it may be worth removing the domains and plans steps here too, as those screen are not used any more.

I also removed the domains & plans steps since they're no longer being used.

>3 .I see my domain is being set to [username][number].wordpress.com when I arrive on the design selection screen, and later on launchpad. I believe that's correct and is now the intended behavior now following the merge of the PR above.

This will be handled in a separate PR.
Two possible solutions:
1. Make a request to `domains/suggestions` API and pass the generated domain to the site creation API call.
2. Modify backend logic for `/new/site` API to use randomly generated domain code for new site creation.

>4. I know you said not to worry about styling, but I figured it may be worth noting issues for now or to handle later. Without checking too carefully, I see the back button is still showing and overlaps the Free page title. Also the progress bar at the top has some space above it, and also doesn't seem to be tracking progress (it is all gray). Are we planning to address styling issues in a later PR?

I updated the styling for the back button and progress bar in this PR since it didn't require too many changes.

File: client/landing/stepper/declarative-flow/free.ts [hunk 1]
```diff
@@ -16,21 +15,16 @@ import {
...
 import Intro from './internals/steps-repository/intro';
 import LaunchPad from './internals/steps-repository/launchpad';
-import PatternsStep from './internals/steps-repository/patterns';
-import PlansStep from './internals/steps-repository/plans';
 import Processing from './internals/steps-repository/processing-step';
 import SiteCreationStep from './internals/steps-repository/site-creation-step';
 import type { Flow, ProvidedDependencies } from './internals/types';
 
 const free: Flow = {
 	name: FREE_FLOW,
```
Comment (edanzer) [inline client/landing/stepper/declarative-flow/free.ts]: Just curious about removing title here. I know the other flow files tend to have this. Do you know when title would be needed vs not?

**Comment by edanzer** [comment]
@agrullon95 - Changes/responses made above look good or make sense. Was going to approve but just found another small issue. I suspect this is related more to the design picker than this PR, but wanted to share and try to confirm that. 

What I'm seeing is that if I go though the flow, and on the design picker screen, choose either Farrow or Heiwa, you are taken to a screen that has the theme preview but has no button to continue or activate or select the theme. So you're stuck on this screen and can't continue. 

Looks like this (note you can't scroll or anything, so what you see here is all there is): 

farrow

I think there's at least one other theme that produces this, but I can't find it again now. Most others I've tested seem to work.

File: client/landing/stepper/declarative-flow/free.ts (hunk 1 above)
Comment (agrullon95) [reply client/landing/stepper/declarative-flow/free.ts]: I added the title logic back and decided to hide the title using CSS. It seems like it's being used for the title in the browser tab and in the header of every screen (next to the WordPress logo): https://github.com/Automattic/wp-calypso/pull/67753. Although there is no code to use the title for the free flow in the browser tab (https://github.com/Automattic/wp-calypso/pull/67753/files#diff-1a5cb229a76dfccdc8b471efedc2ab09f2385c9ba00ae3211ba65452ba1c7680R90), I will leave the title code to keep it consistent with the other "Flows".

**Comment by agrullon95** [comment]
> I think there's at least one other theme that produces this, but I can't find it again now. Most others I've tested seem to work.

@edanzer - Good find. This is a CSS issue related to the themes without style variations. I pushed up a change to fix this issue.

**Comment by edanzer** [comment]
@agrullon95 - Confirmed issue with theme selection pages missing the continue button is fixed. And just generally retested. Everything here tests well for me.

**Review by edanzer — APPROVED** [review]

**Comment by sharpirate** [comment]
✅ I just retested this and it works well with both free and premium themes

**Comment by jeyip** [comment]
## Testing
### Requirements

**Free Theme**
- [x] Can create a site successfully through the free flow
- [x] Theme selected from the design screen is properly registered for the new site ( confirmed in Appearance >  Themes )

**Premium Theme**
- [x] Can create a site successfully through the free flow
- [x] Theme selected from the design screen is properly registered for the new site ( confirmed in Appearance >  Themes )

### Browsers
- [x] Chrome
- [x] Firefox

**Comment by jeyip** [comment]
`NB`: One thing I noticed is that navigating "back" with the browser from the design screen reloads the processing screen, which then redirects to Launchpad with a `null` siteSlug query param.

Wondering if you have thoughts @agrullon95

**Review by jeyip — APPROVED** [review]

**Comment by edanzer** [comment]
> One thing I noticed is that navigating "back" with the browser from the design screen reloads the processing screen, which then redirects to Launchpad with a null siteSlug query param.

I had noticed that, too. Was figuring we could let it pass. Do we think it's worth adding a fix to prevent people from being able to go back via the back button? If we did, I think that could go in a separate PR.

**Comment by jeyip** [comment]
> Do we think it's worth adding a fix to prevent people from being able to go back via the back button

I do -- it seems like a relatively common action folks might take expecting to navigate back to the site setup page, which, unfortunately, would break the rest of the flow

> If we did, I think that could go in a separate PR.

My thoughts exactly 😄 p1671570560252719/1671568521.406149-slack-CHN6J22MP

**Comment by agrullon95** [comment]
> I do -- it seems like a relatively common action folks might take expecting to navigate back to the site setup page, which, unfortunately, would break the rest of the flow

I agree. The back navigation breaks the flow and causes unintended results sometimes (like creating another blog). I'll look into changing this behavior.

**Pull request merged**